sys/sysent.c

```diff
@@ -362,6 +363,21 @@ static int sys_access(thread_t *td, syscall_args_t *args) {
   return result;
 }
 
+static int sys_clock_gettime(thread_t *td, syscall_args_t *args) {
+  clockid_t clk = (clockid_t)args->args[0];
+  timespec_t *ts = (timespec_t *)args->args[1];
```

Comment:
It is **strictly prohibited** to operate on userspace memory from within kernel code. You **must** use `copyin` and `copyout` routines to safely move data between kernelspace and userspace.